### PR TITLE
Merge the PublicAPI checker into main

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Errors at the moment:

- src\Core\src\Platform\iOS\MauiUIApplicationDelegate.cs(118,23,118,36):  
  error RS0016: Symbol 'DidDisconnect' is not part of the declared API
- src\Core\src\Graphics\MauiDrawable.Android.cs(224,15,224,34):  
  error RS0016: Symbol 'SetEmptyBorderBrush' is not part of the declared API
- src\Core\src\PublicAPI\net6.0-windows\PublicAPI.Shipped.txt(1504,1,1504,34):  
  error RS0017: Symbol 'Microsoft.Maui.Platform.RootPanel' is part of the declared API, but is either not public or could not be found
- src\Core\src\PublicAPI\net6.0-windows\PublicAPI.Shipped.txt(1505,1,1505,54):  
  error RS0017: Symbol 'Microsoft.Maui.Platform.RootPanel.RootPanel() -> void' is part of the declared API, but is either not public or could not be found

### Issues Fixed

Fixes #

### CODE REVIEW

File: src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -0,0 +1,6957 @@
...
+~Microsoft.Maui.Controls.Compatibility.Layout<T>.this[int index].get -> Microsoft.Maui.IView
+~Microsoft.Maui.Controls.Compatibility.Layout<T>.this[int index].set -> void
+~Microsoft.Maui.Controls.Compatibility.RelativeLayout.Children.get -> Microsoft.Maui.Controls.Compatibility.RelativeLayout.IRelativeList<Microsoft.Maui.Controls.View>
+~Microsoft.Maui.Controls.Compatibility.RelativeLayout.IRelativeList<T>
+~Microsoft.Maui.Controls.Compatibility.RelativeLayout.IRelativeList<T>.Add(T view, Microsoft.Maui.Controls.Compatibility.Constraint xConstraint = null, Microsoft.Maui.Controls.Compatibility.Constraint yConstraint = null, Microsoft.Maui.Controls.Compatibility.Constraint widthConstraint = null, Microsoft.Maui.Controls.Compatibility.Constraint heightConstraint = null) -> void
+~Microsoft.Maui.Controls.Compatibility.RelativeLayout.IRelativeList<T>.Add(T view, System.Linq.Expressions.Expression<System.Func<double>> x = null, System.Linq.Expressions.Expression<System.Func<double>> y = null, System.Linq.Expressions.Expression<System.Func<double>> width = null, System.Linq.Expressions.Expression<System.F
```
Comment (campersau) [inline src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Shipped.txt:4158]: Would be nice if async APIs could be made consistent by adding an `Async` suffix everywhere.
See also https://github.com/dotnet/maui/issues/6720 (There are more cases than just these.)

Comment (mattleibow) [reply src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Shipped.txt:4158]: Not sure we can change APIs at this point. Maybe we can add an overload, but for a name-only thing?

**Comment by mattleibow** [comment]
Now a new one:

error RS0016: Symbol 'Auto' is not part of the declared API  
error RS0016: Symbol 'FlowDirectionTypeConverter' is not part of the declared API  
error RS0016: Symbol 'implicit constructor for 'FlowDirectionTypeConverter'' is not part of the declared API 
error RS0016: Symbol 'CanConvertFrom' is not part of the declared API   
error RS0016: Symbol 'CanConvertTo' is not part of the declared API  
error RS0016: Symbol 'ConvertFrom' is not part of the declared API  
error RS0016: Symbol 'ConvertTo' is not part of the declared API   
error RS0016: Symbol 'MeasureOverride' is not part of the declared API   
error RS0016: Symbol 'ArrangeOverride' is not part of the declared API   
error RS0017: Symbol 'static Microsoft.Maui.Layouts.FlexBasis.Auto -> Microsoft.Maui.Layouts.FlexBasis' is part of the declared API, but is either not public or could not be found

File: src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs [hunk 2]
```diff
@@ -93,6 +94,24 @@ CGSize UpdateCellSize()
 			return size;
 		}
 
+		[Obsolete]
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		protected void Layout(CGSize constraints)
```
Comment (mattleibow) [inline src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs:99]: @rmarinho you removed this in #7447 and it does not seem to be used. Not sure if adding back as it was is good or I should throw or something?

File: src/Controls/src/Core/Platform/Android/Extensions/FormattedStringExtensions.cs [hunk 3]
```diff
@@ -26,7 +26,28 @@ public static SpannableString ToSpannableString(this Label label)
 				label.TextTransform,
 				label.TextDecorations);
 
+		// TODO: this overload must be removed in net7.0 and replaced with the one below
 		public static SpannableString ToSpannableString(
+			this FormattedString formattedString,
+			IFontManager fontManager,
+			TextPaint? textPaint = null,
```
Comment (mattleibow) [inline src/Controls/src/Core/Platform/Android/Extensions/FormattedStringExtensions.cs:33]: Using TextPaint was incorrect as it did not represent the span size but the label. So the label would have a tiny font size and the span huge, but then layouts would measure line heights using the tine label size.

File: src/Controls/src/Core/Platform/Windows/Extensions/FormattedStringExtensions.cs [hunk 4]
```diff
@@ -25,7 +25,7 @@ public static void UpdateInlines(this TextBlock textBlock, Label label)
 			this TextBlock textBlock,
 			IFontManager fontManager,
 			FormattedString formattedString,
-			double defaultLineHeight = -1d,
+			double defaultLineHeight = 0d, // TODO: should be -1, but too late to change for net6
```
Comment (mattleibow) [inline src/Controls/src/Core/Platform/Windows/Extensions/FormattedStringExtensions.cs]: Not super essential as this is just the default value and was passed in via the call above. This overload really should have been internal.

File: src/Controls/src/Core/PublicAPI/net6.0-android/PublicAPI.Unshipped.txt [hunk 5]
```diff
@@ -1 +1,4 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
+override Microsoft.Maui.Controls.Grid.OnBindingContextChanged() -> void
+~override Microsoft.Maui.Controls.Handlers.Compatibility.ViewCellRenderer.DisconnectHandler(Android.Views.View platformView) -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net6.0-android/PublicAPI.Unshipped.txt:4]: Are overrides really breaking?

File: src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs (hunk 2 above)
Comment (rmarinho) [reply src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs:99]: no it wasn't used anywhere that's why i removed, but is fine to add it back

File: src/Controls/src/Core/PublicAPI/net6.0-android/PublicAPI.Unshipped.txt (hunk 5 above)
Comment (rmarinho) [reply src/Controls/src/Core/PublicAPI/net6.0-android/PublicAPI.Unshipped.txt:4]: Good question, i wouldn't think so.

**Pull request merged**